tests/infra/server.py

```diff
@@ -0,0 +1,37 @@
+import threading
+from http.server import HTTPServer, BaseHTTPRequestHandler
+from typing import Tuple
+
+
+class ThreadedLocalServer(threading.Thread):
+    """
+    Run an http server on a separate thread.
+    """
+    def __init__(self, handler_class: BaseHTTPRequestHandler, address: Tuple[str, int]=('', 8000)):
```

Comment:
Suggestion, it might be worth grabbing an unused port with something like:

```
import contextlib
import socket

def unused_tcp_port():
    with contextlib.closing(socket.socket()) as sock:
        sock.bind(('127.0.0.1', 0))
        return sock.getsockname()[1]
```

Reply:
This is a good suggestion, and I intend to implement it soon. For now, punting to a future PR.